lib/api/core/models/repositories/userRepository.js

```diff
@@ -73,22 +79,22 @@ module.exports = function (kuzzle) {
       Object.assign(data, source);
     }
 
-    return Repository.prototype.hydrate(user, data)
+    return q(_.assignIn(user, data))
```

Comment:
Same here: you can simplify the code a lot by removing the `u` and `result` variables, calling `_.assignIn()` directly without creating a promise, and use the `user` variable thenafter.